### PR TITLE
Homework 18: contact book with server

### PR DESCRIPTION
**Pull request by osechka333**

### CODE REVIEW

File: contact-book-connected-server/server_contact_book.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    for (const input of inputs) {
+        if(Object.hasOwn(contact, input.name)) {
+            input.value = contact[input.name];
+        }
+    }
+}
+function updateKeys(id, newContact) {
+    const oldContact = findContactById(id);
+
+    Object.keys(newContact).forEach(key => oldContact[key] = newContact[key]);
```
Comment (mif83) [inline contact-book-connected-server/server_contact_book.js]: тут можно 
Object.assign(oldContact, newContact)

File: contact-book-connected-server/server_contact_book.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                contactList.push(newContact);
+                renderContact(newContact);
+                clearInput();
+            })
+            .catch(displayErrorMessage);
+    }
+}
+function deleteContact(contact) {
+    if(contact.id) {
+        Contacts.deleteContact(contact.id)
```
Comment (mif83) [inline contact-book-connected-server/server_contact_book.js:130]: забули видалити тут
`let contactList = []`

**Review by mif83 — APPROVED** [review]

File: contact-book-connected-server/server_contact_book.js (hunk 1 above)
Comment (osechka333) [reply contact-book-connected-server/server_contact_book.js]: thanks! Updated to Object.assign(newContact, oldContact);

File: contact-book-connected-server/server_contact_book.js (hunk 2 above)
Comment (osechka333) [reply contact-book-connected-server/server_contact_book.js:130]: fixed!
function deleteContact(contact) {
    if(contact.id) {
        const index = contactList.indexOf(contact);
        Contacts.deleteContact(contact.id)
            .then(() => {
                contactList.splice(index);
        }).catch(displayErrorMessage)
    }
}

**Pull request merged**